src/libasr/pass/intrinsic_functions.h

```diff
@@ -1875,6 +1876,70 @@ namespace Sqrt {
 
 }  // namespace Sqrt
 
+namespace Exponent {
+
+    static ASR::expr_t* eval_Exponent(Allocator& al, const Location& loc,
+            ASR::ttype_t* arg_type, Vec<ASR::expr_t*>& args, diag::Diagnostics& /*diag*/) {
+        ASR::ttype_t* arguement_type = expr_type(args[0]);
+        int32_t kind = extract_kind_from_ttype_t(arguement_type);
+        
+        if (kind == 4) { 
+            float x = ASR::down_cast<ASR::RealConstant_t>(args[0])->m_r;
+            if (x == 0.0) {
+                return make_ConstantWithType(make_IntegerConstant_t, 0, arg_type, loc);
+            }
+            int32_t ix = *reinterpret_cast<int32_t*>(&x);
```

Comment:
```suggestion
            int32_t ix = int32_t(x);
```
https://github.com/lfortran/lfortran/blob/9e7690d58bff665348a654c89329da58cc362e59/src/libasr/pass/intrinsic_functions.h#L1685

Reply:
With this the test fails:

```console
% lfortran ./integration_tests/intrinsics_170.f90 --fast
3
-1022
ERROR STOP
```

Reply:
This line gives the error:
```console
/home/runner/work/lfortran/lfortran/src/libasr/../libasr/pass/intrinsic_functions.h:1891:27: error: dereferencing type-punned pointer will break strict-aliasing rules [-Werror=strict-aliasing]
 1891 |             int32_t ix = *reinterpret_cast<int32_t*>(&x);
      |                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
It looks like one has to do:
```suggestion
            int32_t ix;
            memcpy(&ix, &x, sizeof(ix));
```

Reply:
After C++20, one should be using https://en.cppreference.com/w/cpp/numeric/bit_cast for this it seems. But we are not requiring C++20 yet, so let's use the memcpy trick above. That seems to be the recommended way to do this.